### ISSUE TITLE
[feature request] Delay live updates by set time

### ISSUE THREAD

**Issue by jdetmold** [tinuzz/wp-plugin-trackserver#29]
A nice feature would be an addition to the shortcode for delay = 3d, this could ignore any updates within the past 3d allowing a privacy window for publicly posting location.

**Comment by tinuzz** [comment]
Just to make sure I understand: you want the map to live-update, but with old data? So I'd be watching a track being constantly updated, but with locations that were logged some time ago?

Since this is likely quite easy to implement, I will consider it. But I must admit, it doesn't sound very useful to me.

Why not just publish your tracks a few days after you made them?

Best regards,
Martijn.

**Comment by jdetmold** [comment]
Yup that exactly what I’m thinking. Mostly for security so there’s no interaction but people don’t know where you are right now live. For example I am wanting to use it on my boat so people can follow my progress but they wouldn’t be able to track me down live if they wanted to. Gives me a few days security if that makes sense.

On Apr 7, 2021, at 4:31 PM, Martijn Grendelman ***@***.***> wrote:

﻿

Just to make sure I understand: you want the map to live-update, but with old data? So I'd be watching a track being constantly updated, but with locations that were logged some time ago?

Since this is likely quite easy to implement, I will consider it. But I must admit, it doesn't sound very useful to me.

Why not just publish your tracks a few days after you made them?

Best regards,
Martijn.

**Comment by nickstolwijk** [comment]
I would like to second this feature. I am a leader at Scouting and we want to use it for a game, where other teams see your location, but with a delay. (So it is not too simple to catch you) I would like the delay to be minutes instead of days, so please make that configurable.

**Comment by nickstolwijk** [comment]
That would be great!

For people wondering about the game (in Dutch): http://handboek.pe1mew.nl/?Spelidee%C3%ABn:The_Outlaw_becomes_the_Hunted

**Comment by tinuzz** [comment]
Hmmmm...

Looking at the code, it appears I incorrectly remembered how the `maxage` parameter works. It doesn't actually truncate the track when it is displayed, but instead, it just doesn't display the track (the whole track) if it hasn't been updated for the specified amount of time. If the track _has_ been updated recently enough, it is displayed in full.

So the new parameter should work a little bit differently. Hence, I think using `delay` as the shortcode parameter, as proposed by @jdetmold, would make more sense.

**Comment by nickstolwijk** [comment]
I had need for it tomorrow, so I have tried to add it myself. Looks like it is working! (Normally I don't do PHP but Java, but it looks like I could handle this. ;-) )

**Comment by p3ck** [comment]
This is a great feature!!  Thanks
